File: sample_code.java
public class Calculator {
    /**
     * Recursive factorial calculation
     * Metadata:
     * name: factorial
     * dependencies: [factorial]
     */
    public int factorial(int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }

    /**
     * Performs addition
     * Metadata:
     * name: add
     * dependencies: [validate]
     */
    public int add(int a, int b) {
        if (validate(a) && validate(b)) {
            return a + b;
        }
        return 0;
    }

    /**
     * Validates input
     * Metadata:
     * name: validate
     * dependencies: []
     */
    private boolean validate(int n) {
        return n >= 0;
    }
}

File: sample_code.c
/**
 * Recursive fibonacci implementation
 * Metadata:
 * name: fibonacci
 * dependencies: [fibonacci]
 */
int fibonacci(int n) {
    if (n <= 1) return n;
    return fibonacci(n-1) + fibonacci(n-2);
}

/**
 * Data validation function
 * Metadata:
 * name: validate_data
 * dependencies: [log_error]
 */
int validate_data(int* data, int size) {
    if (!data || size <= 0) {
        log_error("Invalid input");
        return 0;
    }
    return 1;
}

/**
 * Error logging utility
 * Metadata:
 * name: log_error
 * dependencies: []
 */
void log_error(const char* message) {
    printf("Error: %s\n", message);
}